lib/modules/heater/devices/viomi.heater.v4.js

```diff
@@ -0,0 +1,80 @@
+    this.addPropertyByString('heater:status', '{"siid":2,"piid":3,"type":"urn:miot-spec-v2:property:status:00000007:viomi-v4:1","description":"Status","format":"uint8","access":["read","notify"],"unit":"none","valueList":[{"value":0,"description":"Idle"},{"value":1,"description":"Busy"},{"value":2,"description":"Appointment"}]}');
+    this.addPropertyByString('heater:mode', '{"siid":2,"piid":4,"type":"urn:miot-spec-v2:property:mode:00000008:viomi-v4:1","description":"Mode","format":"uint8","access":["read","write","notify"],"unit":"none","valueList":[{"value":1,"description":"Auto"},{"value":2,"description":"Strong"}]}');
+    this.addPropertyByString('heater:target-temperature', '{"siid":2,"piid":5,"type":"urn:miot-spec-v2:property:target-temperature:00000021:viomi-v4:1","description":"Target Temperature","format":"uint16","access":["read","write","notify"],"unit":"celsius","valueRange":[5,35,1]}');
+    this.addPropertyByString('heater:temperature', '{"siid":2,"piid":7,"type":"urn:miot-spec-v2:property:temperature:00000020:viomi-v4:1","description":"Temperature","format":"float","access":["read","notify"],"unit":"celsius","valueRange":[-30,100,1]}');
+    this.addPropertyByString('heater:working-time', '{"siid":2,"piid":8,"type":"urn:miot-spec-v2:property:working-time:00000079:viomi-v4:1","description":"Working Time","format":"uint32","access":["read","notify"],"unit":"minutes","valueRange":[0,99999999,1]}');
+    this.addPropertyByString('countdown:countdown-time', '{"siid":3,"piid":1,"type":"urn:miot-spec-v2:property:countdown-time:00000055:viomi-v4:1","description":"Countdown Time","format":"float","access":["read","write","notify"],"unit":"hours","valueRange":[0,24,0.5]}');
+    this.addPropertyByString('humidifier:on', '{"siid":4,"piid":1,"type":"urn:miot-spec-v2:property:on:00000006:viomi-v4:1","description":"Switch Status","format":"bool","access":["read","write","notify"]}');
+    this.addPropertyByString('humidifier:fault', '{"siid":4,"piid":2,"type":"urn:miot-spec-v2:property:fault:00000009:viomi-v4:1","description":"Device Fault","format":"uint8","access":["read","notify"],"unit":"none","valueList":[{"value":0,"description":"No Faults"}]}');
+    this.addPropertyByString('screen:on', '{"siid":5,"piid":1,"type":"urn:miot-spec-v2:property:on:00000006:viomi-v4:1","description":"Switch Status","format":"bool","access":["read","write","notify"]}');
+    this.addPropertyByString('alarm:alarm', '{"siid":6,"piid":1,"type":"urn:miot-spec-v2:property:alarm:00000012:viomi-v4:1","description":"Alarm","format":"bool","access":["read","write","notify"]}');
+  }
+
+  initDeviceActions() {
+   //no actions
+  }
+
+  initDeviceEvents() {
+    //no events
+  }
+
+
+  /*----------========== VALUES OVERRIDES ==========----------*/
+
+
+  /*----------========== PROPERTY OVERRIDES ==========----------*/
```

Comment:
You could also simply override the temperature property here and return the value of `heater:temperature` instead. Would look like this:

>   temperatureProp() {
>     return this.getProperty('heater:temperature');
>   }

That way you do not need to include code for the specific property handling, since this is more or less the same property, it is just on a different service.